### PR TITLE
chore: improve frontend startup performance

### PR DESCRIPTION
**Pull request by sdirix**
The frontend awaits `initialize`, `configure` and `onStart` for all
frontend contributions. It's therefore very important that no expensive
code is run there.

By not awaiting expensive operations in

 - DebugFrontendApplicationContribution.onStart (~300ms)
 - EditorNavigationContribution.onStart (~36-400ms)
 - TerminalFrontendContribution.onStart (~100-300ms)

the reported startup time without using plugins is reduced by ~400-1000ms
which is an improvement of ~20-40%.

Contributed on behalf of STMicroelectronics

---
Note: I also tried to run all `initialize`, `configure` and `onStart` in "parallel", using `Promise.all`. However when combined with the remaining changes this actually lead to a slowdown, probably because the Javascript thread is already fully utilized.

### CODE REVIEW

**Comment by tsmaeder** [comment]
I was wondering how we are measuring startup time. In the case of a regular electron startup, these changes do not make a noticeable improvement, at least not a second.

**Comment by sdirix** [comment]
In the worst case it's just 300ms faster which is not that noticeable in the overall Electron startup, especially if running in `development` mode which is much slower anyway.

I was simply relying on the internal measurements which you can retrieve when starting the application with `--logLevel=debug`.

File: packages/core/src/browser/common-frontend-contribution.ts [hunk 1]
```diff
@@ -428,11 +428,12 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
     protected pinnedKey: ContextKey<boolean>;
 
     async configure(app: FrontendApplication): Promise<void> {
-        const configDirUri = await this.environments.getConfigDirUri();
-        // Global settings
-        this.encodingRegistry.registerOverride({
-            encoding: UTF8,
-            parent: new URI(configDirUri)
+        this.environments.getConfigDirUri().then(uri => {
```
[the remark this answers is not in the document]
Comment (sdirix) [reply packages/core/src/browser/common-frontend-contribution.ts:431]: Ah I see. We could move the rest of all the methods also in the then

**Comment by sdirix** [comment]
@tsmaeder Are you fine with the current state of the PR or do you see any problems with getting it merged. I would like to open it against upstream.

**Comment by tortmayr** [comment]
I also had a look at this.Everything seems to work as before and I can confirm the reported startup time improvements (based on the stopwatch measurements with `--loglevel debug`).
Also I was trying to identify certain edge cases that could break with the async-> sync changs but could not find anything.

**Comment by sdirix** [comment]
Closed in favor of https://github.com/eclipse-theia/theia/pull/12936

**Pull request closed without merging**